File: ncaa_scraper/ncaa_scraper/sports.py
"""NCAA stats sport configuration."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class SportConfig:
    code: str
    label: str
    default_org_id: int = 164
    divisions: tuple[str, ...] = ("1",)


SPORTS: dict[str, SportConfig] = {
    "MBB": SportConfig("MBB", "Men's Basketball", divisions=("1", "2", "3")),
    "WBB": SportConfig("WBB", "Women's Basketball", divisions=("1", "2", "3")),
    "MBA": SportConfig("MBA", "Baseball", divisions=("1", "2", "3")),
    "WSB": SportConfig("WSB", "Softball", divisions=("1", "2", "3")),
    "MFB": SportConfig("MFB", "Football", divisions=("1", "2", "3")),
    "MSO": SportConfig("MSO", "Men's Soccer", divisions=("1", "2", "3")),
    "WSO": SportConfig("WSO", "Women's Soccer", divisions=("1", "2", "3")),
}


DEFAULT_SPORT_CODE = "MBB"
DEFAULT_SPORT_CODES = ("MBB", "WBB", "MBA", "WSB", "MFB", "MSO", "WSO")


def sport_config(code: str) -> SportConfig:
    normalized = code.upper()
    if normalized not in SPORTS:
        raise ValueError(f"Unsupported NCAA sport code: {code}")
    return SPORTS[normalized]

File: frontend/src/lib/sports.ts
export const SPORTS = [
  { code: "s_mbb", sourceCode: "MBB", label: "Men's Basketball", shortLabel: "MBB", detail: "Men's basketball" },
  { code: "s_wbb", sourceCode: "WBB", label: "Women's Basketball", shortLabel: "WBB", detail: "Women's basketball" },
  { code: "s_bsb", sourceCode: "MBA", label: "Baseball", shortLabel: "BSB", detail: "Baseball" },
  { code: "s_sfb", sourceCode: "WSB", label: "Softball", shortLabel: "SFB", detail: "Softball" },
  { code: "s_fbl", sourceCode: "MFB", label: "Football", shortLabel: "FBL", detail: "Football" },
  { code: "s_mso", sourceCode: "MSO", label: "Men's Soccer", shortLabel: "MSO", detail: "Men's soccer" },
  { code: "s_wso", sourceCode: "WSO", label: "Women's Soccer", shortLabel: "WSO", detail: "Women's soccer" },
];

export const DEFAULT_SPORT = "s_mbb";

export const DIVISIONS = [
  { code: "1", label: "Division I" },
  { code: "2", label: "Division II" },
  { code: "3", label: "Division III" },
] as const;

export function sourceSportCode(sportId: string) {
  return SPORTS.find((sport) => sport.code === sportId)?.sourceCode ?? sportId;
}
